$OpenBSD: patch-makefile,v 1.11 2019/10/30 14:31:07 fcambus Exp $

- Disable hardcoded -O3 optimizations
- Fix GCC_VERSION and CLANG_VERSION detection

Index: makefile
--- makefile.orig
+++ makefile
@@ -208,7 +208,7 @@ GENIEOS := freebsd
 endif
 ifeq ($(firstword $(filter OpenBSD,$(UNAME))),OpenBSD)
 OS := openbsd
-GENIEOS := freebsd
+GENIEOS := bsd
 endif
 ifeq ($(firstword $(filter Darwin,$(UNAME))),Darwin)
 OS := macosx
@@ -589,11 +589,6 @@ PROFILER =
 SYMBOLS = 1
 endif
 
-# specify a default optimization level if none explicitly stated
-ifndef OPTIMIZE
-OPTIMIZE = 3
-endif
-
 # set the symbols level
 ifdef SYMBOLS
 PARAMS += --SYMBOLS='$(SYMBOLS)'
@@ -958,8 +953,8 @@ endif
 
 ifeq ($(OS),windows)
 ifeq (posix,$(SHELLTYPE))
-GCC_VERSION      := $(shell $(TOOLCHAIN)$(subst @,,$(CC)) -dumpversion 2> /dev/null)
-CLANG_VERSION    := $(shell $(TOOLCHAIN)$(subst @,,$(CC)) --version 2> /dev/null| head -n 1 | grep clang | sed "s/^.*[^0-9]\([0-9][0-9]*\.[0-9][0-9]*\.[0-9][0-9]*\).*$$/\1/" | head -n 1)
+GCC_VERSION      := $(shell cc -dumpversion 2> /dev/null)
+CLANG_VERSION    := $(shell cc --version 2> /dev/null | head -n 1 | grep -e 'version [0-9]\.[0-9]\.[0-9]' -o | sed -e 's,version ,,' | tail -n 1)
 PYTHON_AVAILABLE := $(shell $(PYTHON) --version > /dev/null 2>&1 && echo python)
 GIT_AVAILABLE    := $(shell git --version > /dev/null 2>&1 && echo git)
 else
@@ -1014,9 +1009,9 @@ else
 $(info Clang $(CLANG_VERSION) detected)
 ifneq ($(TARGETOS),asmjs)
 ifeq ($(ARCHITECTURE),_x64)
-ARCHITECTURE := _x64_clang
+ARCHITECTURE := _x64
 else
-ARCHITECTURE := _x86_clang
+ARCHITECTURE := _x86
 endif
 endif
 endif
